api/internal/handler/global_rule/global_rule_test.go

```diff
@@ -0,0 +1,338 @@
+		wantErr   error
+		wantInput store.ListInput
+		wantRet   interface{}
+	}{
+		{
+			caseDesc: "list all condition",
+			giveInput: &ListInput{
+				Pagination: store.Pagination{
+					PageSize:   10,
+					PageNumber: 1,
+				},
+			},
+			wantInput: store.ListInput{
+				PageSize:   10,
+				PageNumber: 1,
+			},
+			giveData: []*entity.GlobalPlugins{
+				{ID: "global-rules-1"},
+				{ID: "global-rules-2"},
+				{ID: "global-rules-3"},
+			},
+			wantRet: &store.ListOutput{
+				Rows: []interface{}{
+					&entity.GlobalPlugins{ID: "global-rules-1"},
+					&entity.GlobalPlugins{ID: "global-rules-2"},
+					&entity.GlobalPlugins{ID: "global-rules-3"},
+				},
+				TotalSize: 3,
+			},
+		},
+		{
+			caseDesc: "store list failed",
+			giveInput: &ListInput{
+				Pagination: store.Pagination{
+					PageSize:   10,
+					PageNumber: 10,
+				},
+			},
+			wantInput: store.ListInput{
+				PageSize:   10,
+				PageNumber: 10,
+			},
+			giveData: []*entity.GlobalPlugins{},
+			giveErr:  fmt.Errorf("list failed"),
+			wantErr:  fmt.Errorf("list failed"),
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.caseDesc, func(t *testing.T) {
+			getCalled := true
+			mStore := &store.MockInterface{}
+			mStore.On("List", mock.Anything).Run(func(args mock.Arguments) {
+				getCalled = true
+				input := args.Get(0).(store.ListInput)
+				assert.Equal(t, tc.wantInput.PageSize, input.PageSize)
+				assert.Equal(t, tc.wantInput.PageNumber, input.PageNumber)
+			}).Return(func(input store.ListInput) *store.ListOutput {
+				var returnData []interface{}
+				for _, c := range tc.giveData {
+					if input.Predicate == nil || input.Predicate(c) {
+						returnData = append(returnData, c)
+					}
+				}
+				return &store.ListOutput{
+					Rows:      returnData,
+					TotalSize: len(returnData),
+				}
+			}, tc.giveErr)
+
+			h := Handler{globalRuleStore: mStore}
+			ctx := droplet.NewContext()
+			ctx.SetInput(tc.giveInput)
+			ret, err := h.List(ctx)
+			assert.True(t, getCalled)
+			assert.Equal(t, tc.wantRet, ret)
+			assert.Equal(t, tc.wantErr, err)
+		})
+	}
+}
+
+func TestHandler_Set(t *testing.T) {
+	tests := []struct {
+		caseDesc   string
+		giveInput  *entity.GlobalPlugins
+		giveCtx    context.Context
+		giveErr    error
+		wantErr    error
+		wantInput  *entity.GlobalPlugins
+		wantRet    interface{}
+		wantCalled bool
+	}{
+		{
+			caseDesc: "normal",
+			giveInput: &entity.GlobalPlugins{
+				ID: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{},
+				},
+			},
+			giveCtx: context.WithValue(context.Background(), "test", "value"),
+			wantInput: &entity.GlobalPlugins{
+				ID: "name",
+				Plugins: map[string]interface{}{
+					"jwt-auth": map[string]interface{}{},
+				},
+			},
+			wantRet:    nil,
+			wantCalled: true,
+		},
+		{
+			caseDesc: "store create failed",
+			giveInput: &entity.GlobalPlugins{
+				ID:      "name",
+				Plugins: nil,
+			},
+			giveErr: fmt.Errorf("create failed"),
```

Comment:
What error data will response to client?

Reply:
it's just a mock error here.